homepage/tests/courses_view_tests.py

```diff
@@ -0,0 +1,180 @@
+    courses_list = [
+         Course(1, "Course1", True, 3, "N/A", 1.3, 4, 5, 1),
+         Course(2, "Course2", False, 2, "N/A", 3.5, 2, 13, 5),
+         Course(3, "Course3", False, 1, "N/A", 2, 2.5, 0, 0)
+         ]
+    for course in courses_list:
+        course.save()
+
+    return courses_list
+
+
+@pytest.fixture
+def preqs_list(courses):
+    preqs = [Prerequisites(course_id=courses[1], req_course_id=courses[0], req_code=Prerequisites.Req_Code.BEFORE),
+             Prerequisites(course_id=courses[1], req_course_id=courses[2], req_code=Prerequisites.Req_Code.BEFORE)]
+    for preq in preqs:
+        preq.save()
+    return preqs
+
+
+@pytest.fixture
+def all_courses():
+    return Course.objects.all()
+
+
+# test that each of the new courses is returned with the method
+@pytest.mark.django_db
+def test_get_courses(courses):
+    flag = True
+    db_courses = Course.get_courses()
+    for course in courses:
+        flag = flag and db_courses.filter(pk=course.pk).exists()
+
+    assert flag
+
+
+# test the has_preqs method - if a course with preqs returns True and withuot returns False
+@pytest.mark.django_db
+def test_course_has_preqs(courses, preqs_list):
+    has_preqs = []
+    desired_res = [False, True, False]
+    for course in courses:
+        has_preqs.append(course.has_preqs())
+    for ind in range(0, 2):
+        assert has_preqs[ind] == desired_res[ind]
+
+# -- testing single filter
+
+
+@pytest.mark.django_db
+def test_get_filtered_courses_by_rating(courses, all_courses):
+    # testing for the average rating >= 2.5 which should include Course1 and Course3
+    db_courses = Course.get_filtered_courses_by_rating(2.5, all_courses)
+
+    assert db_courses.filter(pk=courses[0].course_id) and db_courses.filter(pk=courses[2].course_id)
+    assert not db_courses.filter(pk=courses[1].course_id)
+
+
+@pytest.mark.django_db
+def test_get_filtered_courses_by_load(courses, all_courses):
+    # testing for the average load <= 2 which should include Course1 and Course3
+    db_courses = Course.get_filtered_courses_by_load(2, all_courses)
+
+    assert db_courses.filter(pk=courses[0].course_id) and db_courses.filter(pk=courses[2].course_id)
+    assert not db_courses.filter(pk=courses[1].course_id)
+
+
+@pytest.mark.django_db
+def test_get_mandatory_courses(courses, all_courses):
+    # should get back only Course1 from courses
+    db_courses = Course.get_mandatory_courses(all_courses)
+
+    assert db_courses.filter(pk=courses[0].course_id)
+    assert not db_courses.filter(pk=courses[1].course_id) and not db_courses.filter(pk=courses[2].course_id)
+
+
+@pytest.mark.django_db
+def test_get_elective_courses(courses, all_courses):
+    # should get back only Course1 from courses
+    db_courses = Course.get_elective_courses(all_courses)
+
+    assert not db_courses.filter(pk=courses[0].course_id)
+    assert db_courses.filter(pk=courses[1].course_id) and db_courses.filter(pk=courses[2].course_id)
+
+# -- testing double filter
```

Comment:
?

Reply:
like why am I testing a filter over a filter?
I wanted to make sure a user can choose multiple filters and receive a proper result
It may be redundant, I'm not sure yet.

Reply:
No it's not redundant, sorry for the non informative comment it was a long day
I just didn't understand which section the comment is referring to, thought you left it there by mistake